wcivf/apps/elections/tests/test_election_views.py

```diff
@@ -39,3 +45,39 @@ def test_election_detail_view(self):
         self.assertEqual(response.status_code, 200)
         self.assertTemplateUsed(response, "elections/election_view.html")
         self.assertContains(response, self.election.nice_election_name)
+
+
+class ElectionPostViewTests(TestCase):
+    def setUp(self):
+        self.election = ElectionFactory(
+            name="Adur local election",
+            election_date="2021-05-06",
+            slug="local.adur.churchill.2021-05-06",
+        )
+        self.post = PostFactory()
+        self.post_election = PostElectionFactory(
+            election=self.election, post=self.post
+        )
+
+    def test_zero_candidates(self):
+        response = self.client.get(
+            self.post_election.get_absolute_url(), follow=True
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "elections/post_view.html")
+        # self.assertContains(response, "No candidates known yet.")
+
+    def test_num_candidates(self):
+        people = [PersonFactory() for p in range(5)]
+        for person in people:
+            PersonPostFactory(
+                post_election=self.post_election,
+                election=self.election,
+                post=self.post,
+                person=person,
+            )
+
+        response = self.client.get(self.post_election.get_absolute_url())
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "elections/post_view.html")
+        # self.assertContains(response, "The 5 candidates in Adur local election")
```

Comment:
I think this one may fail because of the "in Adur local election" line. In the template, "Adur" is read from the label field on the post object:
```
candidates in {{ object.post.label }}
```
And looking in the PostFactory is hardcoded to "Copeland". You can update your `PostFactory` definition in setUp to specify "Adur" as the label to use:

Reply:
Or alternatively, build the string you asserting against with the `Post` object that is created:
```
self.assertContains(response, f"The 5 candidates in {self.post.label}")
```

Reply:
I like this, thanks 👍